website/static/js/addonNodeConfig.js

```diff
@@ -0,0 +1,162 @@
+ * @param {String} url API url to initially fetch settings
+ * @param {String} selector CSS selector for containing div
+ * @param {String} folderPicker CSS selector for folderPicker div
+ * @param {Object} opts Optional overrides to the class' default treebeardOptions, in particular onPickFolder
+ */
+var AddonFolderPickerViewModel = oop.extend(FolderPickerViewModel, {
+    constructor: function(addonName, url, selector, folderPicker, opts) {
+        var self = this;
+        self.super.constructor.call(self, addonName, url, selector, folderPicker);
+        // whether the auth token is valid
+        self.validCredentials = ko.observable(true);
+        // Emails of contributors, can only be populated by activating the share dialog
+        self.emails = ko.observableArray([]);
+        self.loading = ko.observable(false);
+        // Whether the contributor emails have been loaded from the server
+        self.loadedEmails = ko.observable(false);
+        // externalAccounts
+        self.accounts = ko.observable([]);
+        // List of contributor emails as a comma-separated values
+        self.emailList = ko.pureComputed(function() {
+            return self.emails().join([', ']);
+        });
+        self.selectedFolderType = ko.pureComputed(function() {
+            var userHasAuth = self.userHasAuth();
+            var selected = self.selected();
+            return (userHasAuth && selected) ? selected.type : '';
+        });
+        self.messages.SUBMIT_SETTINGS_SUCCESS =  ko.pureComputed(function() {
+            return 'Successfully linked "' + $osf.htmlEscape(self.folder().name) + '". Go to the <a href="' +
+                self.urls().files + '">Files page</a> to view your content.';
+        });
+        // Overrides
+        var defaults = {
+            onPickFolder: function(evt, item) {
+                evt.preventDefault();
+                var name = item.data.path !== '/' ? item.data.path : '/ (Full ' + self.addonName + ')';
+                self.selected({
+                    name: name,
+                    path: item.data.path,
+                    id: item.data.id
+                });
+                return false; // Prevent event propagation
+            },
+            connectAccount: function() {
+                window.location.href = this.urls().auth;
+            }
+        };
+        // Overrides
+        self.options = $.extend({}, defaults, opts);
+        // Treebeard config
+        self.treebeardOptions = $.extend(
+            {}, 
+            FolderPickerViewModel.prototype.treebeardOptions,
+            {
+                onPickFolder: function(evt, item) {
+                    return this.options.onPickFolder.call(this, evt, item);
+                }.bind(this),
+                resolveLazyloadUrl: function(item) {
+                    return item.data.urls.folders;
+                }
+            }
+        );
+    },
+    toggleShare: function() {
```

Comment:
This "sharing" code is no longer necessary due to https://github.com/CenterForOpenScience/osf.io/issues/2310 and should be removed.